core/types/receipt.go

```diff
@@ -55,13 +56,20 @@ type Receipt struct {
 	TxHash          common.Hash    `json:"transactionHash" gencodec:"required"`
 	ContractAddress common.Address `json:"contractAddress"`
 	GasUsed         uint64         `json:"gasUsed" gencodec:"required"`
+
+	// Block location fields - not part of the rlp
+	BlockHash        common.Hash `json:"blockHash"`
+	BlockNumber      *big.Int    `json:"blockNumber"`
+	TransactionIndex *big.Int    `json:"transactionIndex"`
 }
 
 type receiptMarshaling struct {
 	PostState         hexutil.Bytes
 	Status            hexutil.Uint64
 	CumulativeGasUsed hexutil.Uint64
 	GasUsed           hexutil.Uint64
+	BlockNumber       *hexutil.Big
+	TransactionIndex  *hexutil.Big
```

Comment:
I'm not too certain about this -- doesn't this mean that you change the RLP-representation?

Reply:
[`// receiptRLP is the consensus encoding of a receipt.`](https://github.com/ethereum/go-ethereum/pull/17662/files/5d0009093b1dd601069c18d6f4defc1547559ea0#diff-cf3bf12023ca997cd6b43322a70ea69dR75) so `receiptMarshaling` is just for struct->json.

Reply:
I can conform that receiptMarshaling just exists for JSON purposes.